### PR TITLE
Navigation Editor: Improve experience for user creating their first menu

### LINKED ISSUE #28124: [Navigation Screen] Broken first landing experience when no menus are present.
## Description

When first landing on the Navigation Screen, if the user has no menus they're presented with a screen that has the message 'No menus available'.

However, the user can still edit and save a menu without being given a way to name the menu.

## Step-by-step reproduction instructions
1. Enable the navigation experiment
2. Delete all of your menus (either on the old menus screen, or the new navigation screen).
3. Open up the new navigation screen
4. Observe that there's a menu present that can be edited

## Expected behaviour
Either:
- The user has to explicitly create and name a menu before being able to edit and save (this is consistent with how the screen generally works, but we'd need to decide what to display on screen when the user has no menus).

Or:
- The user can edit their navigation and is prompted to name the menu before saving (I think this is an ok option, but it's different to how menus are named in all other cases).

Or something else entirely (this needs some design thoughts).

In tandem with this, we might want to think about what the user's first experience on this screen should be, how do we make it clear what the first step should be?

## Actual behaviour
When there are no menus present, the user can still edit a navigation block in the editing area of the screen. The user can edit and save it, but has no way to name it, assign it to a location, or delete it (so this is very bad bug!).

## Screenshots or screen recording (optional)
Screenshot 2021-01-15 at 3 09 28 pm

### PR DESCRIPTION
**Pull request by talldan**
## Description
Closes #28124

The flow for users creating a first menu has been less than idea or even somewhat broken for a while. This updates the flow to follow the designs in #28124.

It doesn't update the navigation block placeholder as shown in those designs, but that can come in a later PR.

There are a few areas where our component system doesn't quite match the designs
- The Card component has top/bottom padding that's different to its left/right padding.
- The TextInput component has help text that's kinda close to the input.

We may want to update these component styles separately.

## How has this been tested?
e2e tests updated for this feature.

1. Visit the navigation screen
2. Delete all menus if you have any
3. Create your first menu

## Screenshots 
create-first-menu

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-navigation/src/components/layout/index.js [hunk 1]
```diff
@@ -51,6 +59,9 @@ export default function Layout( { blockEditorSettings } ) {
 
 	useMenuNotifications( selectedMenuId );
 
+	const hasMenus = menus?.length;
```
Comment (kevin940726) [inline packages/edit-navigation/src/components/layout/index.js:62]: Nit: explicitly cast it to boolean to match the same code in header.
```suggestion
	const hasMenus = menus?.length > 0;
```

File: packages/edit-navigation/src/components/layout/index.js [hunk 2]
```diff
@@ -51,6 +59,9 @@ export default function Layout( { blockEditorSettings } ) {
...
+	const isBlockEditorReady = !! hasMenus && navigationPost;
```
Comment (kevin940726) [inline packages/edit-navigation/src/components/layout/index.js:63]: Nit: Should probably wrap it?
```suggestion
	const isBlockEditorReady = !! ( hasMenus && navigationPost );
```

File: packages/edit-navigation/src/store/selectors.js [hunk 3]
```diff
@@ -25,6 +25,10 @@ import { KIND, POST_TYPE, buildNavigationPostId } from './utils';
  */
 export const getNavigationPostForMenu = createRegistrySelector(
 	( select ) => ( state, menuId ) => {
+		if ( menuId === null ) {
```
Comment (kevin940726) [inline packages/edit-navigation/src/store/selectors.js:28]: Why is this needed? Could we instead avoid calling this selector at all when `menuId` is null? Or maybe add a brief comment above this condition :)

Comment (talldan) [reply packages/edit-navigation/src/store/selectors.js]: True, 632437b does that 😄

File: packages/edit-navigation/src/components/layout/index.js (hunk 1 above)
Comment (talldan) [reply packages/edit-navigation/src/components/layout/index.js]: Yep, that was an oversight. 2f04009 makes it a bool.

File: packages/edit-navigation/src/components/layout/index.js (hunk 2 above)
Comment (talldan) [reply packages/edit-navigation/src/components/layout/index.js]: 👍 I've applied that suggestion.

**Review by draganescu — APPROVED** [review]
This works great and the code looks good! 🚢  it!

**Pull request merged**